client/src/components/Collections/common/CollectionEditView.vue

```diff
@@ -0,0 +1,126 @@
+                    <suitable-converters-tab :suitable-converters="item" @clicked-convert="clickedConvert" />
+                </b-tab>
+            </SuitableConvertersProvider>
+        </b-tabs>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import axios from "axios";
+import { prependPath } from "utils/redirect";
+import _l from "utils/localization";
+import { errorMessageAsString } from "utils/simple-error";
+import DatabaseEditTab from "./DatabaseEditTab";
+import SuitableConvertersTab from "./SuitableConvertersTab";
+import { GenomeProvider, SuitableConvertersProvider } from "../../WorkflowInvocationState/providers";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faDatabase, faTable, faBars, faUser, faCog } from "@fortawesome/free-solid-svg-icons";
+
+library.add(faDatabase, faTable, faBars, faUser, faCog);
+
+Vue.use(BootstrapVue);
+export default {
+    created() {
+        this.getCollectionDataAndAttributes();
+    },
+    components: {
+        DatabaseEditTab,
+        SuitableConvertersTab,
+        FontAwesomeIcon,
+        GenomeProvider,
+        SuitableConvertersProvider,
+    },
+    data: function () {
+        return {
+            attributes_data: {},
+            errorMessage: null,
+            jobError: null,
+        };
+    },
+    props: {
+        collection_id: {
+            type: String,
+            required: true,
+        },
+    },
+    computed: {
+        databaseKeyFromElements: function () {
+            return this.attributes_data.dbkey;
+        },
+    },
+    methods: {
+        l(str) {
+            // _l conflicts private methods of Vue internals, expose as l instead
+            return _l(str);
+        },
+        getCollectionDataAndAttributes: async function () {
+            let attributesGet = this.$store.getters.getCollectionAttributes(this.collection_id);
+            if (attributesGet == null) {
+                await this.$store.dispatch("fetchCollectionAttributes", this.collection_id);
+                attributesGet = this.$store.getters.getCollectionAttributes(this.collection_id);
+            }
+            this.attributes_data = attributesGet;
+        },
+        clickedSave: function (attribute, newValue) {
+            const url = prependPath("/api/dataset_collections/" + this.collection_id);
+            const data = {};
+            if (attribute == "dbkey") {
+                data["dbkey"] = newValue.id;
+            } else if (attribute == "file_ext") {
+                data["file_ext"] = newValue.id;
+            }
+            axios.put(url, data).catch(this.handleError);
+        },
+        clickedConvert: function (selectedConverter) {
+            const url = prependPath(`/api/tools/${selectedConverter.tool_id}/convert`);
+            const data = {
+                src: "hdca",
```

Comment:
I think you'll want to include the history id here (could be the global current history id for a start)

Reply:
If you click on an HDCA to edit and then go to a new history in another tab - do you want to take that? I think the HDCA's history is actually a more stable choice than relying on the current history.

Reply:
Galaxy.currentHistory wouldn't currently be shared across tabs like that unless I missed something major.

Reply:
I think you're right @dannon, that's why I think this is better than relying on trans.history

Reply:
I think the better default is the HDCA's history id which is what you'll get if you leave it off - not ``trans.history`` - if y'all prefer it to be some global variable that is okay with me though I'm not going to fight you on it.

Reply:
Well, I was mainly just commenting specifically on the hypothetical there.  Is there any way this gets run into a different history?  If not, just enforcing that it's converted 'in place' using the same history is fine with me.  `trans.history` is not reliable.

Reply:
Yeah, the hdca's history id is fine with me too ... I guess I made it more confusing by calling the currentHistoryId global ? I think that is probably more future proof that the hdca.history_id (e.g if we add drag-and-drop into the input field), but that works too.